lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -140,6 +144,97 @@ class AddNewPrimUndoableCommand : public Ufe::UndoableCommand
     PXR_NS::TfToken _primToken;
 };
 
+const char* selectUSDFileScript = R"(
+global proc string SelectUSDFileForAddReference()
+{
+    string $result[] = `fileDialog2 
+        -fileMode 1
+        -caption "Add Reference to USD Prim"
+        -fileFilter "USD Files (*.usd *.usda *.usdc);;*.usd;;*.usda;;*.usdc"`;
+
+    if (0 == size($result))
+        return "";
+    else
+        return $result[0];
+}
+SelectUSDFileForAddReference();
+)";
+
+const char* clearAllReferencesConfirmScript = R"(
```

Comment:
For the file dialog to choose a reference and a confirm box when clearing I'm just adding a quick and dirty MEL script as a string.  Design has been thinking about what are final design should look like, but this should be good enough to start iterating with them on it.